pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -456,18 +472,27 @@ func (s *statsSyncLoad) updateCachedItem(item model.TableItemID, colHist *statis
 	}
 	if !item.IsIndex && colHist != nil {
 		c, ok := tbl.Columns[item.ID]
-		if !ok || c.IsFullLoad() {
+		// If we don't have this column or its stats is fully loaded, we skip it.
+		if (!ok && !tbl.ColAndIdxExistenceMap.Has(item.ID, false)) || (ok && c.IsFullLoad()) {
 			return true
 		}
 		tbl = tbl.Copy()
-		tbl.Columns[c.ID] = colHist
+		tbl.Columns[item.ID] = colHist
+		// If the column is analyzed we refresh the map for the possible change.
+		if colHist.StatsAvailable() {
+			tbl.ColAndIdxExistenceMap.InsertCol(item.ID, colHist.Info, true)
```

Comment:
I think this is not necessary, right?

Reply:
From the newly added test, i think it's yes.